cpssEnabler/linuxNoKernelModule/drivers/dmaDriver.c

```diff
@@ -28,320 +28,929 @@ disclaimer.
 * dmaDriver.c
 *
 * DESCRIPTION:
-*       mvDmaDrv - driver to map DMA memory to userspace
-*                  Usage:
-*                     fd=open("/dev/mvDmaDrv",O_RDWR);
-*                     mmap(,size, ..,fd,0) will allocate DMA block and map it
-*                     ...
-*                     read(fd,uint64_t*,8) will read DMA address
-*                                          (prvExtDrvDmaPhys64)
-*                     close(fd) will unmap and free DMA block
+*	mvDmaDrv - driver to map DMA memory to userspace
+*	Usage:
+*		fd = open("/dev/mvDmaDrv",O_RDWR);
+*		mmap(,size, ..,fd,0) will allocate DMA block and map it
+*		...
+*		read(fd,uint64_t*,8) will read DMA address
+*		close(fd) will unmap and free DMA block
 *
-*       Additional features used by linuxNokernelModule driver:
-*           1. When virtual address == LINUX_VMA_DMABASE (defined below)
-*              then allocate a single DMA block which will be mapped to
-*              all applications to the same virtual address
-*           2. Select PCI device for allocation:
-*                   lseek(fd,
-*                       ((domain<<16)&0xffff0000)
-*                       |((bus<<8)   &0x0000ff00)
-*                       |((dev<<3)   &0x000000f8)
-*                       |((func)     &0x00000007))
-*               This should allow IOMMU transactions from PCI device to
-*               system memory
+*	Additional features used by linuxNokernelModule driver:
+*		1. When virtual address == LINUX_VMA_DMABASE (defined below)
+*		   then allocate a single DMA block which will be mapped to all
+*		   applications to the same virtual address
+*		2. Select PCI device for allocation:
+*			lseek(fd,
+*			      ((domain<<16)&0xffff0000) |
+*			      ((bus<<8)   &0x0000ff00)  |
+*			      ((dev<<3)   &0x000000f8)  |
+*			      ((func)     &0x00000007))
+*		   This should allow IOMMU transactions from PCI device to
+*		   system memory
 *
-*       Please note:
-*           on Intel CPU it may require 'intel_iommu=off' kernel option
+*	Please note:
+*		On Intel CPU it may require 'intel_iommu=off' kernel option
+*
+*
+*	Following is an example of pre-allocation in device-tree.
+*
+*	reserved-memory {
+*		prestera_rsvd: buffer@0M {
+*			compatible = "shared-dma-pool";
+*			no-map;
+*			reg = <0x0 0x10000 0x0 0x1000000>;
+*		};
+*	};
+*
+*	mvdma {
```

Comment:
Does this pertain to the M0 platform?
I don't see this in our device tree.

Reply:
No, it does not.